pandas/core/groupby/generic.py

```diff
@@ -1334,9 +1345,8 @@ def _transform_general(
                 if cache_key not in NUMBA_FUNC_CACHE:
                     NUMBA_FUNC_CACHE[cache_key] = numba_func
                 # Return the result as a DataFrame for concatenation later
-                res = self.obj._constructor(
-                    res, index=group.index, columns=group.columns
-                )
+                indexer = self._get_index(name) if self.dropna else group.index
```

Comment:
So the groups are constructed in this `BaseGrouper` method:
https://github.com/pandas-dev/pandas/blob/361166f86da1a665a1b5808e0935c28f8cf56d34/pandas/core/groupby/ops.py#L116-L128
I _think_ that `BaseGrouper` doesn't have a `dropna` attribute - I checked inside this method with `hasattr(self, 'dropna')` to be sure. Would the idea be to propagate `dropna` to `BaseGrouper` so that `group.index` is correct?

Not sure this is the way to go but thought I'd ask